### PR TITLE
Improve min stake amount

### LINKED ISSUE #43: Improve min stake amount
The TellorFlex contract ensures that the minimum stake amount is 100 TRB or 1,500 USD by regularly updating the minimum stake amount based on TRB/USD prices submitted to the oracle.

The functionality in https://github.com/tellor-io/tellorFlex/blob/3b3820f2111ec2813cb51455ef68cf0955c51674/contracts/TellorFlex.sol#L351 should therefore be added to the pallet, ideally auto-adjusting the StakeAmount storage item based on a hook run every X blocks. It can also be called manually via a `update_stake_amount` dispatchable function. The runtime should be able to configure the pallet with a queryId for some TRB to other currency rate, along with a min amount in some other currency (e.g. USD, EUR, GBP etc). It might be best to group these related fields into a struct rather then a load of individual config items, if practical.

More information and context in https://github.com/tellor-io/tellor-pallet/issues/17, which is to be tackled as a follow on from this issue.

### PR DESCRIPTION
**Pull request by evilrobot-01**
- adds `MinimumStakeAmount` config item, which is used by `StakeAmount<T>` storage item until value set
- adds `StakeAmountCurrencyTarget` config item, for setting stake amount target in some currency (e.g. USD 1,500)
- adds `StakingTokenPriceQueryId` config item, for obtaining rate for converting between TRB and currency from oracle
- adds `UpdateStakeAmountInterval` config item, for configuring how frequently the stake amount is updated via hook
  - hook compares current time vs `LastStakeAmountUpdate<T>  + interval`, where interval is configured interval or some minimum interval
- adds `update_stake_amount` dispatchable function, updating the stake amount based on oracle price and stake amount currency target
- removes `stake_amount` from `register` dispatchable function as no longer required

Closes #43

### CODE REVIEW

**Review by oraclown — APPROVED** [review]
Thanks for these updates, @evilrobot-01 ! Looks good

File: src/impls.rs [hunk 1]
```diff
@@ -1093,6 +1093,43 @@ impl<T: Config> Pallet<T> {
 		// }
 		Ok(())
 	}
+
+	// Updates the stake amount after retrieving the latest token price from oracle.
+	pub(super) fn _update_stake_amount() -> DispatchResult {
```
Comment (pawanbisht62) [inline src/impls.rs:1098]: I guess, we should not use underscore in the beginning of the function's name. Can we please remove it if you agree on this?

**Review by pawanbisht62 — APPROVED** [review]
LGTM, just added one comment

File: src/impls.rs (hunk 1 above)
Comment (evilrobot-01) [reply src/impls.rs:1098]: There is a clash with the dispatchable function of the same name unfortunately.

I am not sure what the convention is in Rust, maybe do_, but _ is used in the Tellor smart contracts for internal functions, hence the reason why I used it for those that clash within impls.rs.

Any suggestions?

**Comment by evilrobot-01** [comment]
Rebased

File: src/impls.rs (hunk 1 above)
Comment (pawanbisht62) [reply src/impls.rs:1285]: As per my understanding, in Rust we use underscore when the function is not being used anywhere.
Yes, it'd be good if we use do_ just to make our code align with Rust standards.

Comment (evilrobot-01) [reply src/impls.rs:1285]: Will add a final commit 👍

**Pull request merged**